formatter_test.go

```diff
@@ -176,6 +177,41 @@ var gosyntax = []test{
     },
 }`,
 	},
+	{(*time.Time)(nil), "(*time.Time)(nil)"},
+	{&ValueGoString{"vgs"}, `VGS vgs`},
+	{(*ValueGoString)(nil), `(*pretty.ValueGoString)(nil)`},
+	{(*VGSWrapper)(nil), `(*pretty.VGSWrapper)(nil)`},
+	{&PointerGoString{"pgs"}, `PGS pgs`},
+	{(*PointerGoString)(nil), "(*pretty.PointerGoString)(nil)"},
+	{&PanicGoString{"oops!"}, "(*pretty.PanicGoString)(PANIC=GoString method: oops!)"},
```

Comment:
So it looks like to my earlier comment, you do at least get the type here, which points you in the right direction.
It also does match what Format does, though I struggled a lot to get a failure there:
```
package main

import (
    "fmt"
    "time"

    "github.com/kr/pretty"
)

type Value struct {}
func (Value) Format(f fmt.State, verb rune) { f.Write([]byte("formatted"))}
func (Value) GoString() string { return "gostringed" }

type Panic struct {}
func (Panic) Format(f fmt.State, verb rune) { panic("failed")}
func main() {
    now := time.Now()
    var t *time.Time
    // t = &now
    _ = now
    fmt.Println(pretty.Sprint(t))
    fmt.Println("done")
    var val Value
    fmt.Println(val)
    var ptr *Value
    fmt.Println(ptr)
    var pnc Panic
    fmt.Println(pnc)
}
```

ends up outputting:
```
$ go run main.go
%!v(PANIC=Format method: value method time.Time.GoString called using nil *Time pointer)
done
formatted
<nil>
%!v(PANIC=Format method: failed)
```

Its a bit of a shame that the standard is so hard to parse. I'm willing to go along with established precedent, but I do feel that:
```
"(*pretty.PanicGoString)(PANIC=calling method "GoString": oops!)"}
```
Is easier for a human to understand and realize what they need to fix.

Thoughts?

Reply:
Thanks @jameinel. I agree that `calling method "GoString": ...` is easier to read and understand, and that the error message being exactly the same as `fmt`'s is not necessary. Updated!